modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/connector/PetStoreSourceRetryPolicyProviderTestCase.java

```diff
@@ -77,6 +80,32 @@ public void retryPolicySourceFailWithConnectionException() throws Exception {
         }));
   }
 
+  @Test
+  public void retryPolicySourceFailOnStartFallbackToConnection() throws Exception {
+    exception.expect(RetryPolicyExhaustedException.class);
+    exception.expectCause(sameInstance(connectionException));
+    try {
+      startFlow("source-fail-on-start-fallback-to-connection");
+    } catch (Exception e) {
+      new PollingProber(TIMEOUT_MILLIS, POLL_DELAY_MILLIS)
```

Comment:
This can be more compact
```
probe(TIMEOUT_MILLIS, POLL_DELAY_MILLIS, () -> { 
assertThat(PetStoreConnector.timesStarted, is(2));
return true})
```